File: C++/Task300.cpp
#include <iostream>
using namespace std;

int main()
{
    int visits[24]; //оглошуємо масив
    cout<<"Ведіть відвідуваність сайту ";
    cout<<"кожної години протягом доби";
    //вводимо елементи масиву в циклі
    for (int i = 0; i < 24; i++)
    {
        cin >> visits[i];
    }
    int s = 0;
    //цикл по елементах списку
    for (int i = 0; i < 24; i++)
    {
        s += visits[i];
    }
    double a = double(s) / 24;
    printf("Середньодобова відвідуваність %10.2f", a);
    cout<<"Задайте коефіцієнт відхилення у % :";
    double p;
    cin>>p;
    printf("Години з менш, ніж %5.2f -відсотковим", p);
    printf("відхиленням від сереньої відвідуваності:");
    for (int i = 0; i < 24; i++)
    {
        if (abs(visits[i] - a) < a * p / 100)
        {// Години номеруємо 1..24, тому i+1
            cout << i + 1 << endl;
        }
    }
}

File: Java/Task300.java
import java.util.Scanner;

public class Task300 {
    public static void main(String[] args){
        //оглошуємо масив та створюємо його
        int [] visits = new int [24];
        System.out.println("Ведіть відвідуваність " +
                "сайту кожної години протягом доби");
        Scanner input = new Scanner(System.in);
        //вводимо елементи масиву в циклі
        for (int i = 0; i < 24; i++){
            visits[i] = input.nextInt();
        }
        int s = 0;
        //цикл по елементах списку
        for (int i = 0; i < 24; i++){
            s += visits[i];
        }
        double a = (double)s / 24;
        //приводимо s до дійсного типу
        //інакше буде виконуватися ділення націло
        System.out.println("Добова відвідуваність " +
                a + " відвідувачів на годину.");
        System.out.println("Задайте відхилення у %");
        // оголошену змінну ініціалізуємо даними,
        // отриманими з консолі
        double p  = input.nextDouble();
        System.out.println("Години з менш, ніж "+ p +
                "-відсотковим відхиленням від " +
                "середньої відвідуваності:");
        for (int i = 0; i < 24; i++){
            if(Math.abs(visits[i] - a) < a * p / 100) {
                // години номеруємо 1..24
                System.out.printf(" %3d", i + 1);
            }
        }
        input.close();
    }
}



